openhands/integrations/provider.py

```diff
@@ -407,6 +408,111 @@ async def get_branches(
 
         return main_branches + other_branches
 
+    async def get_microagents(self, repository: str) -> list[MicroagentResponse]:
```

Comment:
If we're fetching microagent for a repo would we know which provider its from in advance?